libs/experimental/langchain_experimental/graph_transformers/llm.py

```diff
@@ -158,6 +159,97 @@ def map_to_base_relationship(rel: Any) -> Relationship:
     )
 
 
+def _parse_and_clean_json(
+    argument_json: Dict[str, Any],
+) -> Tuple[List[Node], List[Relationship]]:
+    nodes = []
+    for node in argument_json["nodes"]:
+        if not node.get("id"):  # Id is mandatory, skip this node
+            continue
+        nodes.append(
+            Node(
+                id=node["id"].title(),
+                type=node.get("type").capitalize() if node.get("type") else None,
+            )
+        )
+    relationships = []
+    for rel in argument_json["relationships"]:
+        # Mandatory props
+        if (
+            not rel.get("source_node_id")
+            or not rel.get("target_node_id")
+            or not rel.get("type")
+        ):
+            continue
+
+        # Node type copying if needed from node list
+        if not rel.get("source_node_type"):
+            try:
+                rel["source_node_type"] = [
+                    el["type"]
+                    for el in argument_json["nodes"]
+                    if el["id"] == rel["source_node_id"]
+                ][0].capitalize()
+            except IndexError:
+                rel["source_node_type"] = None
+        if not rel.get("target_node_type"):
+            try:
+                rel["target_node_type"] = [
+                    el["type"]
+                    for el in argument_json["nodes"]
+                    if el["id"] == rel["target_node_id"]
+                ][0].capitalize()
+            except IndexError:
+                rel["target_node_type"] = None
+
+        source_node = Node(
+            id=rel["source_node_id"].title(),
+            type=rel["source_node_type"],
+        )
+        target_node = Node(
+            id=rel["target_node_id"].title(),
+            type=rel["target_node_type"],
+        )
+        relationships.append(
+            Relationship(
+                source=source_node,
+                target=target_node,
+                type=rel["type"].replace(" ", "_").upper(),
```

Comment:
just curious: if we pass in graph schema, would we validate against that after the processing or not at all? (i.e. valid node and rel-types and directions?)